lib/medera/minion/connection.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Medera.Minion.Connection do
+  @moduledoc """
+  Connects to the master node and registers this minion
+
+  Automatically detects when the master goes down and reconnects
+  """
+
+  alias Medera.Minion
+  alias Medera.Minion.Registry
+
+  require Logger
+
+  use GenServer
+
+  @doc "Start in a supervision tree"
+  @spec start_link() :: GenServer.on_start
+  def start_link do
+    name = "minion-" <> (Node.self() |> Atom.to_string)
+    GenServer.start_link(__MODULE__, [], name: {:global, name})
+  end
+
+  def init(_) do
+    {:ok, :disconnected, 10}
```

Comment:
Those are timeouts - see https://hexdocs.pm/elixir/GenServer.html#c:handle_call/3 .  In this particular example, after 10 milliseconds the GenServer will receive a `:timeout` message which gets handled by `handle_info(:timeout, ...)` below.  It's a way to basically implement a state machine (see also http://erlang.org/doc/man/gen_fsm.html).

I should make module attributes to give them names though.